compaction/compaction_manager.cc

```diff
@@ -842,12 +848,15 @@ future<> compaction_manager::drain() {
 }
 
 future<> compaction_manager::stop() {
+    if (_compaction_module) {
+        co_await _compaction_module->stop();
```

Comment:
Can it be stopped multiple times?
If not, better exchange(_compaction_module, nullptr) above to ensure it's stopped exactly once